virtcontainers/api.go

```diff
@@ -760,3 +761,100 @@ func PauseContainer(sandboxID, containerID string) error {
 func ResumeContainer(sandboxID, containerID string) error {
 	return togglePauseContainer(sandboxID, containerID, false)
 }
+
+// AddInterface is the virtcontainers add interface entry point.
+func AddInterface(sandboxID string, inf *grpc.Interface) (*grpc.Interface, error) {
+	if sandboxID == "" {
+		return nil, errNeedSandboxID
+	}
+
+	lockFile, err := rwLockSandbox(sandboxID)
+	if err != nil {
+		return nil, err
+	}
+	defer unlockSandbox(lockFile)
+
+	s, err := fetchSandbox(sandboxID)
+	if err != nil {
+		return nil, err
+	}
+	return s.AddInterface(inf)
+}
+
+// RemoveInterface is the virtcontainers remove interface entry point.
+func RemoveInterface(sandboxID string, inf *grpc.Interface) (*grpc.Interface, error) {
```

Comment:
Again, this is almost identical to `AddInterface()` so you could make them both call a common function like the following to avoid duplication:

```
func toggleInterface(add bool, sandboxID string, inf *grpc.Interface) (*grpc.Interface, error)
```

Reply:
Updated